filestore/README.md

```diff
@@ -0,0 +1,207 @@
+# Notes on the Filestore Datastore
+
+The filestore is a work-in-progress datastore that stores the unixfs
+data component of blocks in files on the filesystem instead of in the
+block itself.  The main use of the datastore is to add content to IPFS
+without duplicating the content in the IPFS datastore.
+
+The filestore is developed on Debian (GNU/Linux).  It has been tested on
+Windows and should work on MacOS X and other Unix like systems.
+
+## Quick start
+
+To add a file to IPFS without copying, use `filestore add -P` or to add a
+directory use `filestore add -P -r`.  (Throughout this document all
+command are assumed to start with `ipfs` so `filestore add` really
+mains `ipfs filestore add`).  For example to add the file `hello.txt`
+use:
+```
+  ipfs filestore add -P hello.txt
+```
+The file or directory will then be added.  You can now try to retrieve
+it from another node such as the ipfs.io gateway.
+
+Paths stores in the filestore must be absolute.  You can either
+provide an absolute path or use one of `-P` (`--physical`) or -l
+(`--logical`) to create one.  The `-P` (or `--physical`) means to make
+a absolute path from the physical working directory without any
+symbolic links in it; the -l (or `--logical`) means to use the `PWD`
+env. variable if possible.
+
+If adding a file with the daemon online the same file must be
+accessible via the path provided by both the client and the server.
+Without extra options it is currently not possible to add directories
+with the daemon online.
+
+If the contents of an added file have changed the block will become
+invalid.  By default, the filestore uses the modification-time to
+determine if a file has changed.  If the mod-time of a file differs
+from what is expected the contents of the block are rechecked by
+recomputing the multihash and failing if the hash differs from what is
+expected.
+
+Adding files to the filestore will generally be faster than adding
+blocks normally as less data is copied around.  Retrieving blocks from
+the filestore takes about the same time when the hash is not
+recomputed, when it is retrieval is slower.
+
+## Adding all files in a directory
+
+The example script in filestore/examples/add-dir.sh can be used to add
+all files in a directly to the filestore and keep the filestore in
+sync with what is the directory.  Just specify the directory you want
+to add or update.  The first time it is run it will add all the files
+in the directory.  When run again it will readd any modified files.  A
```

Comment:
There is evidence for "readd" being a word: http://www.dictionary.com/browse/readd, https://en.wiktionary.org/wiki/readd.  I would rather leave it in unless there is a strong objection.  I can change "readd" to "re-add" if you think that is more correct through.